base/tuple.jl

```diff
@@ -208,15 +208,15 @@ eltype(t::Type{<:Tuple}) = _compute_eltype(t)
 function _tuple_unique_fieldtypes(@nospecialize t)
     @_total_meta
     types = IdSet()
-    t´ = unwrap_unionall(t)
     # Given t = Tuple{Vararg{S}} where S<:Real, the various
     # unwrapping/wrapping/va-handling here will return Real
     if t isa Union
-        union!(types, _tuple_unique_fieldtypes(rewrap_unionall(t´.a, t)))
-        union!(types, _tuple_unique_fieldtypes(rewrap_unionall(t´.b, t)))
+        union!(types, _tuple_unique_fieldtypes(rewrap_unionall(t.a, t)))
+        union!(types, _tuple_unique_fieldtypes(rewrap_unionall(t.b, t)))
     else
         r = Union{}
-        for ti in (t´::DataType).parameters
+        t´ = unwrap_unionall(t)::DataType
```

Comment:
UnionAll can also be wrapped in Union

Reply:
`Union` can be wrapped in `UnionAll`, right?